contracts/modules/delegates/IexecOrderManagementDelegate.sol

```diff
@@ -119,4 +121,43 @@ contract IexecOrderManagementDelegate is IexecOrderManagement, DelegateBase, Sig
             emit ClosedRequestOrder(requestorderHash);
         }
     }
+
+    function computeDealVolume(
+        IexecLibOrders_v5.AppOrder calldata appOrder,
+        IexecLibOrders_v5.DatasetOrder calldata datasetOrder,
+        IexecLibOrders_v5.WorkerpoolOrder calldata workerpoolOrder,
+        IexecLibOrders_v5.RequestOrder calldata requestOrder
+    ) external view override returns (uint256 volume) {
+        return
+            _computeVolume(
+                appOrder.volume,
+                _toTypedDataHash(appOrder.hash()),
+                datasetOrder.dataset != address(0),
+                datasetOrder.volume,
+                _toTypedDataHash(datasetOrder.hash()),
+                workerpoolOrder.volume,
+                _toTypedDataHash(workerpoolOrder.hash()),
+                requestOrder.volume,
+                _toTypedDataHash(requestOrder.hash())
+            );
+    }
+
+    function _computeVolume(
```

Comment:
Why having different names (`computeVolume` & `computeDealVolume`)?

Reply:
I can use `predictFutureDealVolume` for the public function, don't feel like use `_predictFutureDealVolume` for the internal one since the return value : volume will be use to reduce the consumed volume for each Orders. So far from the idea of "predict" ?

Reply:
You are right for internal. So why not `_computeDealVolume` here for the private one? (trying to understand the logic)

Reply:
Seems good to me yes